File: MATLAB/Display RGB Channels.m
function rgb_channels_demo()
    RGB = imread('Enter your image path here'); 
    R = RGB(:,:,1); % Red channel
    G = RGB(:,:,2); % Green channel
    B = RGB(:,:,3); % Blue channel
    zero_channel = zeros(size(R), 'uint8');
    RedChannelImage = cat(3, R, zero_channel, zero_channel); % Red channel in color
    GreenChannelImage = cat(3, zero_channel, G, zero_channel); % Green channel in color
    BlueChannelImage = cat(3, zero_channel, zero_channel, B); % Blue channel in color
    figure;
    subplot(2, 2, 1);
    imshow(RGB);
    title('Original RGB Image', 'FontSize', 12);
    subplot(2, 2, 2);
    imshow(RedChannelImage);
    title('Red Channel', 'FontSize', 12);
    subplot(2, 2, 3);
    imshow(GreenChannelImage);
    title('Green Channel', 'FontSize', 12);
    subplot(2, 2, 4);
    imshow(BlueChannelImage);
    title('Blue Channel', 'FontSize', 12);
end

File: Python/Display RGB Channels.py
import numpy as np
import matplotlib.pyplot as plt
import imageio.v2 as imageio

img = imageio.imread('Enter your image path here')

R = img[:, :, 0]
G = img[:, :, 1]
B = img[:, :, 2]

zero_channel = np.zeros_like(R)

red_channel_img = np.stack((R, zero_channel, zero_channel), axis=2)
green_channel_img = np.stack((zero_channel, G, zero_channel), axis=2)
blue_channel_img = np.stack((zero_channel, zero_channel, B), axis=2)

plt.figure(figsize=(10, 8))

plt.subplot(2, 2, 1)
plt.imshow(img)
plt.title('Original RGB Image', fontsize=12)
plt.axis('off')

plt.subplot(2, 2, 2)
plt.imshow(red_channel_img)
plt.title('Red Channel', fontsize=12)
plt.axis('off')

plt.subplot(2, 2, 3)
plt.imshow(green_channel_img)
plt.title('Green Channel', fontsize=12)
plt.axis('off')

plt.subplot(2, 2, 4)
plt.imshow(blue_channel_img)
plt.title('Blue Channel', fontsize=12)
plt.axis('off')

plt.tight_layout()
plt.show()
